src/AbuseIpDb.php

```diff
@@ -27,25 +30,38 @@ class AbuseIpDb
      */
     protected $days = 30;
 
+    /**
+     * @var $spam_threshold abuse confidence score between 0 (not confident it is a spam ip) and 100 (totally confident)
+     */
+    protected $spam_threshold = 100;
+
+    /**
+     * @var $cache_ttl Cache duration in seconds
+     */
+    protected $cache_ttl = 10;
+
     /**
      * @return bool
-     * @throws \Exception
+     * @throws AbuseIpDbException
      */
-    public function IsSpamIp()
+    public function IsSpamIp($ip = null)
```

Comment:
What is the reason to use/implement this compared with the current setIp? Better (work)-flow?

Reply:
Yes. But that is just my point of view.